### ISSUE TITLE
4.75bpw pack is missing the entire non-expert backbone (attn/router/hc/shared) — word salad root cause; fix included

### ISSUE THREAD

**Issue by fattchris** [vcruz305/DeepSeek-V4.1-Flash-EXL3-DGX-Spark-recipe#13]
## Summary

The 32-shard pack contains **only** routed experts (EXL3), engram tables, the 3 MTP draft layers, vision/aligner globals, and embed/head/norm. The 40 main layers' **non-expert tensors are absent**: attention projections (`wq_a`, `wq_b`, `wkv`, `wo_a`, `wo_b`), compressor, indexer, router (`ffn.gate.*`), shared experts (`ffn.shared_experts.*`), all six `hc_*` hyper-connection tensors per layer, and the per-layer norms.

The README says *"Protected non-expert tensors: Copied (attn / shared / DSpark / vision / head), not wholesale-requantized"* — but they are not in the shards or the index.

## Evidence

Index sha256 `f60c776bac0b53fafd5a0b8670f4cfd1f6697a708f0b4c4e85c97500bacba854` (byte-identical to the repo's `model.safetensors.index.json`):

| category | keys |
|---|---|
| routed experts (`layers.*.ffn.experts.*`) | 186,624 |
| engram | 12 |
| mtp (complete: attn+hc+ffn) | 97 |
| vision/aligner/image | 266 |
| globals (embed/head/norm) | 3 |
| **main-layer non-expert (attn/gate/hc/shared/norms)** | **0** |

Shard-1 header read directly confirms: `embed.weight`, `head.weight`, `norm.weight`, `aligner.*`, `mtp.0/1/2.*` (complete) — no `layers.N.attn.*` anywhere.

### Runtime consequence (4× DGX Spark GB10, TP4+EP4, your vllm-exl3 @ 814d4fe)

With the missing params left at `torch.empty` init:
- attention sublayer output RMS = **0** at every layer (probe-instrumented)
- router logits std = **0** — every token routes to expert 1 (`local_frac=0` on 3 of 4 ranks → those ranks compute nothing)
- shared experts output = 0; `hc_*` = 0; final `cos(h, embed) ≈ 0.05`
- deterministic word salad from token 1 ("The capital of France is" → `ican544 traced Cath/lic出现过gets…`), **byte-identical** output when expert values change (75%-zero-padded vs real) because nothing downstream of the missing weights is reachable

This is likely why GB10 parity was never validated: the pack cannot serve correctly on any platform.

## Fix (verified working)

Pull the missing 1,243 tensors from `deepseek-ai/DeepSeek-V4.1-Flash` (fp8 e4m3 + `[32,32]` ue8m0 block scales — matches the pack config's `non_routed_quantization`, byte-compatible), write as a supplementary `model-00033-of-00033.safetensors` (6.88 GB), extend the index:

```python
pull = {k for k in base_weight_map
        if re.fullmatch(r"layers\.(\d+)\.(.+)", k) and 0 <= int(m.group(1)) < n_layers
        and not m.group(2).startswith(("ffn.experts.", "engram."))}
```

After merging + reload: **"The capital of France is" → "Paris."** (logprob −5e−06), 17×23=391 with shown work, `s[::-1]` — model fully coherent. Probe: attention RMS 0.71, router std 2.0, shared+routed both contributing.

Happy to send the extraction script as a PR to the recipe repo if useful.
